Lib/multiprocessing/shared_memory.py

```diff
@@ -73,6 +73,8 @@ class SharedMemory:
     def __init__(self, name=None, create=False, size=0):
         if not size >= 0:
             raise ValueError("'size' must be a positive integer")
+        if create and size == 0:
```

Comment:
IMHO, raising ValueError need to be discussed with core developers.

Reply:
Actually, if `size == 0 and create is True`, then the shared memory segment is created but a `ValueError` is raised by `mmap`  [here](https://github.com/python/cpython/blob/0e2c856e74d8702d5903ff9f5a6fbbaa436b2bf3/Lib/multiprocessing/shared_memory.py#L113).
So, previously also, ValueError was being raised. But, shared memory segment was being created and it wasn't unlinked.

I could have added `except ValueError` at [L114](https://github.com/python/cpython/blob/0e2c856e74d8702d5903ff9f5a6fbbaa436b2bf3/Lib/multiprocessing/shared_memory.py#L113) with `except OSError` to call `self.unlink` and the raise.
But I think that the implemented approach in PR would be better, since this way there wouldn't be any hassle of linking and unlinking the shared memory, if invalid create size is being passed.